lib/rules/disallow-newline-before-block-statements.js

```diff
@@ -199,6 +203,11 @@ module.exports.prototype = {
     }
 };
 
+function isBareBlock(node) {
+    return node.parentNode &&
+    node.parentNode.type === 'BlockStatement' || node.parentNode.type === 'Program';
```

Comment:
What if there's a bare block in a function declaration like this:
```js
function foo() {
  {
    let key = 'bar';
  }
  // no more key
}
```

Wouldn't the parentNode be != Program then and it'd be the same issue?